solutions/swb-app/src/datasetRoutes.ts

```diff
@@ -85,6 +87,25 @@ export function setUpDSRoutes(
     })
   );
 
+  // Delete dataset
+  router.delete(
+    '/datasets/:datasetId',
+    wrapAsync(async (req: Request, res: Response) => {
+      const environments: Environment[] = await environmentService.listEnvironmentsForDataSet(
+        req.params.datasetId
+      );
+      if (environments.length) {
+        const existingEnvironments = environments.map((env) => `${env.name}(${env.id})`).join(', ');
+        throw Boom.badRequest(
+          `Cannot delete dataset because it is associated with environments: ${existingEnvironments}`
+        );
+      }
+
+      await dataSetService.removeDataSet(req.params.datasetId);
```

Comment:
Thoughts on passing in a `checkDependency` function into `removeDataset`. That way we can ensure that no one will accidentally use `dataSetService.removeDataSet` incorrectly. 

See [here](https://github.com/aws-solutions/solution-spark-on-aws/blob/feat/costCenter/solutions/swb-app/src/costCenterRoutes.ts#L53) for an example.

Reply:
We can do that, however you can still do `.removeDataset(id, () => {});` and avoid checks. The only way to enforce it is to validate all business rules in dataset service

Reply:
@bhudnell are you ok with us adding a `checkDependency` param to the method signature of `removeDataset`. It forces the implementer to think about whether any dependent resource needs to be deleted before a dataset is removed. We're using a similar approach [here](https://github.com/aws-solutions/solution-spark-on-aws/pull/641#discussion_r1028583222).

Reply:
We discussed offline, MAF is fine with the change. Please make a separate PR for it, merging into develop and be sure to include proper integ and unit testing